src/test_graph.py

```diff
@@ -4,15 +4,16 @@
 
 TABLE = [(
     [1, 2, 3],
```

Comment:
You should have a larger range of cases than this. Empty graph, graph with only one item, graph with nodes but no edges, graph with 100s of edges, etc. There are some flaws in your graph that your tests are not digging up.